app/test_main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+import pytest
+
+from unittest import mock
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "valid_url,has_connection,expected",
+    [
+        pytest.param(True, True, "Accessible",
+                     id="valid url and has connection"),
+        pytest.param(True, False, "Not accessible",
+                     id="valid url and no connection"),
+        pytest.param(False, True, "Not accessible",
+                     id="invalid url and has connection"),
+        pytest.param(False, False, "Not accessible",
+                     id="invalid url and no connection"),
+    ]
+)
+def test_can_access_google_page(
+        valid_url: bool,
+        has_connection: bool,
+        expected: str
+) -> None:
+    with (
+        mock.patch("app.main.valid_google_url") as mocked_valid_url,
+        mock.patch("app.main.has_internet_connection") as mocked_connection
+    ):
```

Comment:
You could use mocks as decorators to not allocate a context manager. Please, reformat your code to follow this approach